src/inmanta/data/__init__.py

```diff
@@ -5681,10 +5734,19 @@ async def get_increment(
                 ):
                     in_increment = True
                     break
+
             if in_increment:
                 increment.append(resource)
             else:
-                next.append(resource)
+                # If no events, shortcut on known good
+                if (
+                    status == const.ResourceState.deployed.name
+                    and resource["attribute_hash"] == resource["last_deployed_attribute_hash"]
+                ):
+                    # Shortcut on easy good ones
+                    not_increment.append(resource)
```

Comment:
I'm not convinced this is actually the same. What if another version is deploying at the moment? The previous implementation would inspect that version before ever getting to the last deployed one, and decide there that it should be in the increment if the hash does not match.

Here, the hash matches, but the intermediate deploying version might have a different hash.

Overall, I see how these changes are good opportunities, but unless they are really trivial, I would prefer to split them off into a separate PR to keep the scope contained.